tests/_test_mscolab/test_models.py

```diff
@@ -55,3 +71,83 @@ def test_verify_password(self):
         user = User(self.userdata[0], self.userdata[1], self.userdata[2])
         assert user.verify_password("fail") is False
         assert user.verify_password(self.userdata[2]) is True
+
+
+class Test_Permission:
+    @pytest.fixture(autouse=True)
+    def setup(self):
+        self.u_id = 1
+        self.op_id = 1
+        self.access_level = "admin"
+
+    def test_permission_creation(self):
+        permission = Permission(self.u_id, self.op_id, self.access_level)
+
+        assert permission.u_id == self.u_id
+        assert permission.op_id == self.op_id
+        assert permission.access_level == self.access_level
```

Comment:
While other tests are structured this way, I would like to not introduce new classes when they are not required (and they are never actually required). This test could just be:
```python
def test_permission_creation(self):
    permission = Permission(1, 1, "admin")

    assert permission.u_id == 1
    assert permission.op_id == 1
    assert permission.access_level == "admin"
```
and be more legible.

The `setup` autouse fixtures are more of a crutch to convert old unittest-style setup and teardown methods of classes into pytest-compatible fixtures, without larger changes to the tests. Fixtures should be reserved for setup and teardown of resources that a test might require, like setting up a database connection. If a test really requires a fixture and that fixture is shared between multiple tests I would prefer if those tests still were plain functions and each of them requested the fixture themselves, instead of using a class with an autouse fixture.

This comment applies to the other (newly introduced) classes as well.

Reply:
Now you have also moved the pre-existing tests in the Test_User class, with the effect that the setup autouse fixture now applies to all tests in the file. This is not what I meant. What I meant is that the newly added tests should be at the top-level, but the Test_User class with the tests that you haven't added (and the autouse fixture) should stay as they were (refactoring those could be a future effort over the entire test suite).